math/src/field/fields/fft_friendly/quartic_babybear.rs

```diff
@@ -0,0 +1,552 @@
+            &a[0] * &b[3] + &a[3] * &b[0] + &a[1] * &b[2] + &a[2] * &b[1],
+        ]
+    }
+
+    fn square(a: &Self::BaseType) -> Self::BaseType {
+        [
+            &a[0].square() - BETA * ((&a[1] * &a[3]).double() + &a[2].square()),
+            (&a[0] * &a[1] - BETA * (&a[2] * &a[3])).double(),
+            (&a[0] * &a[2]).double() + &a[1].square() - BETA * (&a[3].square()),
+            (&a[0] * &a[3] + &a[1] * &a[2]).double(),
+        ]
+    }
+
+    fn sub(a: &Self::BaseType, b: &Self::BaseType) -> Self::BaseType {
+        [&a[0] - &b[0], &a[1] - &b[1], &a[2] - &b[2], &a[3] - &b[3]]
+    }
+
+    fn neg(a: &Self::BaseType) -> Self::BaseType {
+        [-&a[0], -&a[1], -&a[2], -&a[3]]
+    }
+
+    /// Return te inverse of a fp4 element if exist.
+    /// This algorithm is inspired by Risc0 implementation:
+    /// <https://github.com/risc0/risc0/blob/4c41c739779ef2759a01ebcf808faf0fbffe8793/risc0/core/src/field/baby_bear.rs#L460>
+    fn inv(a: &Self::BaseType) -> Result<Self::BaseType, FieldError> {
+        let mut b0 = &a[0] * &a[0] + BETA * (&a[1] * (&a[3] + &a[3]) - &a[2] * &a[2]);
+        let mut b2 = &a[0] * (&a[2] + &a[2]) - &a[1] * &a[1] + BETA * (&a[3] * &a[3]);
+        let c = &b0.square() + BETA * b2.square();
+        let c_inv = c.inv()?;
+        b0 *= &c_inv;
+        b2 *= &c_inv;
+        Ok([
+            &a[0] * &b0 + BETA * &a[2] * &b2,
+            -&a[1] * &b0 - BETA * &a[3] * &b2,
+            -&a[0] * &b2 + &a[2] * &b0,
+            &a[1] * &b2 - &a[3] * &b0,
+        ])
+    }
+
+    fn div(a: &Self::BaseType, b: &Self::BaseType) -> Self::BaseType {
+        <Self as IsField>::mul(a, &Self::inv(b).unwrap())
+    }
+
+    fn eq(a: &Self::BaseType, b: &Self::BaseType) -> bool {
+        a[0] == b[0] && a[1] == b[1] && a[2] == b[2] && a[3] == b[3]
+    }
+
+    fn zero() -> Self::BaseType {
+        [
+            FieldElement::zero(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+        ]
+    }
+
+    fn one() -> Self::BaseType {
+        [
+            FieldElement::one(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+        ]
+    }
+
+    fn from_u64(x: u64) -> Self::BaseType {
+        [
+            FieldElement::from(x),
+            FieldElement::zero(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+        ]
+    }
+
+    /// Takes as input an element of BaseType and returns the internal representation
+    /// of that element in the field.
+    /// Note: for this case this is simply the identity, because the components
+    /// already have correct representations.
+    fn from_base_type(x: Self::BaseType) -> Self::BaseType {
+        x
+    }
+
+    fn double(a: &Self::BaseType) -> Self::BaseType {
+        <Degree4BabyBearExtensionField as IsField>::add(a, a)
+    }
+
+    fn pow<T>(a: &Self::BaseType, mut exponent: T) -> Self::BaseType
```

Comment:
This can be written simpler. There are some repetitions of code

Reply:
Solved  [here](https://github.com/lambdaclass/lambdaworks/pull/942/commits/18def0a2a17ed6c9b50518fdd744625d2c39e850)